hw_5/HW_pandas_matplotlib.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import math
+import matplotlib.ticker as tck
+
+
+# In[2]:
+
+
+plt.rcParams['font.size'] = '20'
+plt.rcParams['figure.figsize'] = (20, 10)
+
+
+# # Task1
+
+# In[3]:
+
+
+def read_gff(path):
+    df = pd.read_csv(path, delimiter='\t', header=1, names=['chromosome', 'source','type', 'start', 'end', 'score', 'strand', 'phase', 'attributes'])
+    return df
+
+
+# In[4]:
+
+
+def read_bed6(path):
+    df = pd.read_csv(path, delimiter='\t', names=['chromosome', 'start_bed', 'end', 'name', 'score', 'strand'])
+    return df
+
+
+# In[5]:
+
+
+gff = read_gff('data/rrna_annotation.gff')
+gff['attributes'] = gff['attributes'].str[5:8].str.replace('_','')
+gff
+
+
+# In[6]:
+
+
+bed = read_bed6('data/alignment.bed')
+bed
+
+
+# In[7]:
+
+
+rRNA_count = gff.groupby(['chromosome', 'attributes'], as_index=False).size()
+rRNA_count.columns = ['Sample', 'rRNA type', 'Count']
+rRNA_count
+
+
+# In[8]:
+
+
+plt.subplots(figsize = (20,10))
+sns.barplot(data = rRNA_count, x = 'Sample', hue = 'rRNA type', y = 'Count')
+plt.xticks(rotation=90)
+
+
+# In[9]:
+
+
+intersect = gff.merge(bed, on = 'chromosome', suffixes = ('_gff', '_bed')).drop_duplicates()
+intersect_rRNA = intersect.query('start > start_bed & end_gff < end_bed')
+intersect_rRNA
+
+
+# # Task2
+
+# In[10]:
+
+
+df = pd.read_csv('data/diffexpr_data.tsv.gz', delimiter='\t')
+df
+
+
+# In[11]:
+
+
+sd = df.query('logFC < 0 & pval_corr < 0.05')
+su = df.query('logFC > 0 & pval_corr < 0.05')
+nd = df.query('logFC < 0 & pval_corr >= 0.05')
+nu = df.query('logFC > 0 & pval_corr >= 0.05')
```

Comment:
Не надо жалеть символы на имена переменных
```suggestion
significantly_downregulated_genes = df.query('logFC < 0 & pval_corr < 0.05')
su = df.query('logFC > 0 & pval_corr < 0.05')
nd = df.query('logFC < 0 & pval_corr >= 0.05')
nu = df.query('logFC > 0 & pval_corr >= 0.05')
```